pkgs/by-name/ud/udev-hid-bpf/package.nix

```diff
@@ -0,0 +1,91 @@
+{
+  bpftools,
+  clang,
+  elfutils,
+  fetchFromGitLab,
+  lib,
+  libbpf,
+  meson,
+  ninja,
+  pkg-config,
+  python3,
+  python3Packages,
+  rustPlatform,
+  systemd,
+  zlib,
+}:
+rustPlatform.buildRustPackage rec {
```

Comment:
This is a meson project, not a cargo one, even though it contains a cargo project.

See <https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md#rust-package-built-with-meson-rust-package-built-with-meson>. This should fix a bunch of the hacks you have below with `buildPhase` and whatever